jdaviz/tests/test_subsets.py

```diff
@@ -857,3 +858,93 @@ def test_delete_subsets(cubeviz_helper, spectral_cube_wcs):
+        cubeviz_helper.app.get_viewer('flux-viewer').apply_roi(CircularROI(30, 30, 10))
+
+        subsets = cubeviz_helper.app.get_subsets(include_sky_region=True)
+        assert len(subsets['Subset 1']) == 2
+        sky_region_0 = subsets['Subset 1'][0]['sky_region']
+        sky_region_1 = subsets['Subset 1'][1]['sky_region']
+        assert_allclose(sky_region_0.center.ra.deg, 24.40786313)
+        assert_allclose(sky_region_0.center.dec.deg, 22.45185308)
+        assert_allclose(sky_region_0.radius.arcsec, 28001.08106569353)
+        assert_allclose(sky_region_1.center.ra.deg, 28.41569583)
+        assert_allclose(sky_region_1.center.dec.deg, 25.44814949)
+        assert_allclose(sky_region_1.radius.arcsec, 25816.498273)
+
+        # and that they are both None when not specified
+        subsets = cubeviz_helper.app.get_subsets()
+        assert subsets['Subset 1'][0]['sky_region'] is None
+        assert subsets['Subset 1'][1]['sky_region'] is None
+
+    def test_get_regions_from_subsets_imviz(self, imviz_helper, spectral_cube_wcs):
+
+        """ Basic tests for retrieving Sky Regions from subsets in Imviz.
+        """
+
+        wcs = spectral_cube_wcs.celestial
+        data = NDData(np.ones((128, 128)) * u.nJy, wcs=wcs)
+        imviz_helper.load_data(data)
+
+        # basic test, a single circular region
+        imviz_helper.app.get_viewer('imviz-0').apply_roi(CircularROI(25, 25, 10))
+        subsets = imviz_helper.app.get_subsets(include_sky_region=True)
+        sky_region = subsets['Subset 1'][0]['sky_region']
+        assert isinstance(sky_region, CircleSkyRegion)
+        assert_allclose(sky_region.center.ra.deg, 24.40786313)
+        assert_allclose(sky_region.center.dec.deg, 22.45185308)
+        assert_allclose(sky_region.radius.arcsec, 28001.08106569353)
+
+        # now test a composite subset, each component should have a sky region
+        imviz_helper.app.session.edit_subset_mode.mode = AndMode
+        imviz_helper.app.get_viewer('imviz-0').apply_roi(CircularROI(30, 30, 10))
+
+        subsets = imviz_helper.app.get_subsets(include_sky_region=True)
+        assert len(subsets['Subset 1']) == 2
+        sky_region_0 = subsets['Subset 1'][0]['sky_region']
+        sky_region_1 = subsets['Subset 1'][1]['sky_region']
+        assert_allclose(sky_region_0.center.ra.deg, 24.40786313)
+        assert_allclose(sky_region_0.center.dec.deg, 22.45185308)
+        assert_allclose(sky_region_0.radius.arcsec, 28001.08106569353)
+        assert_allclose(sky_region_1.center.ra.deg, 28.41569583)
+        assert_allclose(sky_region_1.center.dec.deg, 25.44814949)
+        assert_allclose(sky_region_1.radius.arcsec, 25816.498273)
+
+    def test_no_wcs_sky_regions(self, imviz_helper):
+
+        """ Make sure that if sky regions are requested and there is no WCS,
+            that it returns None with no error.
+        """
+
+        data = NDData(np.ones((128, 128)) * u.nJy)
```

Comment:
Can reduce the size a little since you don't draw the circle that large?
```suggestion
        data = NDData(np.ones((40, 40)) * u.nJy)
```

Reply:
done.